stagemonitor-core/src/main/java/org/stagemonitor/core/configuration/RemotePropertiesConfigurationSource.java

```diff
@@ -0,0 +1,112 @@
+ * to pull the appropriate configuration. <p> Example as returned from the server when requesting *.properties:
+ * <pre> {@code stagemonitor.active: true
+ * stagemonitor.instrument.include: my.domain
+ * stagemonitor.web.paths.excluded: /some/path
+ * }</pre>
+ * The RemotePropertiesConfigurationSource is controlled via the following properties. See {@code CorePlugin} for
+ * their corresponding detailed description
+ * <pre>
+ * stagemonitor.configuration.remoteproperties.urls=configUrl[, configUrl]...
+ * stagemonitor.configuration.springcloud.deactivateStagemonitorIfConfigServerIsDown=(true|false)
+ * </pre>
+ */
+public class RemotePropertiesConfigurationSource extends AbstractConfigurationSource {
+	private static final Logger logger = LoggerFactory.getLogger(RemotePropertiesConfigurationSource.class);
+
+	private Properties properties;
+	private HttpClient httpClient;
+	private String configUrl;
+
+	/**
+	 * Creates an RemotePropertiesConfigurationSource instance and fetches the configuration initially via a http GET
+	 *
+	 * @param configUrl Http or https address of the config server
+	 */
+	public RemotePropertiesConfigurationSource(String configUrl) {
+		this(new HttpClient(), configUrl);
+	}
+
+	/**
+	 * Creates an RemotePropertiesConfigurationSource instance and fetches the configuration initially via a http GET
+	 *
+	 * @param httpClient Instance of HttpClient
+	 * @param configUrl  Http or https address of the config server
+	 */
+	public RemotePropertiesConfigurationSource(HttpClient httpClient, String configUrl) {
+
+		if (StringUtils.isEmpty(configUrl) || !configUrl.startsWith("http")) {
+			throw new IllegalArgumentException("Invalid config server address: " + configUrl);
+		}
+
+		this.httpClient = httpClient;
+		this.properties = new Properties();
+		this.configUrl = configUrl;
+
+		reload();
+	}
+
+	/**
+	 * Return the property value for a given key
+	 *
+	 * @param key the property key
+	 * @return value of property, or null
+	 */
+	@Override
+	public String getValue(String key) {
+		return properties.getProperty(key);
+	}
+
+	/**
+	 * Returns the config url of this config source
+	 *
+	 * @return full config url
+	 */
+	@Override
+	public String getName() {
+		return configUrl;
+	}
+
+	@Override
+	public void reload() {
+
+		logger.debug("Requesting configuration from: " + configUrl);
+
+		httpClient.send("GET", configUrl, new HashMap<String, String>(), null, new HttpClient.ResponseHandler<Void>() {
+			@Override
+			public Void handleResponse(HttpRequest<?> httpRequest, InputStream is, Integer statusCode, IOException e) throws IOException {
+				if (e != null || statusCode != 200 || is == null) {
+					logger.warn("Couldn't GET configuration. " + configUrl + " returned " + statusCode);
+					return null;
+				}
+
+				logger.debug("Reloading RemotePropertiesConfigurationSource with: " + properties);
+				properties.clear();
```

Comment:
That could create a small time window where there are no properties

Reply:
Fixed by assigning a new Properties instance with the updated configuration